storagemarket/lp2pimpl/deal_stream.go

```diff
@@ -0,0 +1,53 @@
+package lp2pimpl
+
+import (
+	"bufio"
+
+	"github.com/filecoin-project/boost/storagemarket/types"
+
+	"github.com/libp2p/go-libp2p-core/host"
+	"github.com/libp2p/go-libp2p-core/mux"
+	"github.com/libp2p/go-libp2p-core/peer"
+
+	cborutil "github.com/filecoin-project/go-cbor-util"
+)
+
+// DealStream reads and writes deal proposals to a libp2p stream
+type DealStream struct {
+	p        peer.ID
+	host     host.Host
+	rw       mux.MuxedStream
+	buffered *bufio.Reader
```

Comment:
You can get away without having this var as the stream is also an io.Reader. If you wanna use a buffered reader over the stream, increase the size to 4K.